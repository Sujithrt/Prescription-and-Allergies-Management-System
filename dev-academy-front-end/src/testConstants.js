export const patient = {
    "id": 1001,
    "firstName": "Leanne",
    "lastName": "Graham",
    "gender": "Male",
    "age": 21,
    "phoneNumber": 9123456780,
    "email": "leanne.graham@gmail.com",
    "city": "Bangalore",
    "state": "Karnataka",
    "medicalConditions": [
        {
            "id": 1,
            "name": "Cephalalgia (headache)"
        },
        {
            "id": 3,
            "name": "Otitis externa (swimmer's ear)"
        },
        {
            "id": 9,
            "name": "Contusion (bruise)"
        }
    ],
    "allergies": [
        {
            "id": 7,
            "name": "Aspirin Allergy",
            "reaction": "Tiredness",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 4,
                "name": "Aspirin"
            },
            "encounter": {
                "id": 10,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 8,
            "name": "Paracetamol",
            "reaction": "sdfd",
            "severity": "MILD",
            "status": "ACTIVE",
            "ingredient": {
                "id": 1,
                "name": "Paracetamol"
            },
            "encounter": {
                "id": 13,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 13,
            "name": "Ibuprofin Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 20,
                "date": "2021-10-29",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 14,
            "name": "Paracetamol Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 1,
                "date": "2021-10-26",
                "patient": {
                    "id": 1003,
                    "firstName": "Clementine",
                    "lastName": "Bauch",
                    "gender": "Female",
                    "age": 56,
                    "phoneNumber": 9123456782,
                    "email": "clementine.bauch@gmail.com",
                    "city": "Mysore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 23,
                            "name": "Gestation (pregnancy) "
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        }
    ]
};

export const user = {
    "id": 1,
    "username": "John",
    "email": "john.cage@gmail.com",
    "role": "Physician",
    "firstName": "John",
    "lastName": "Cage",
    "speciality": "Cardiologist",
    "city": "Bangalore",
    "state": "Karnataka",
    "dob": "1975-06-12T18:30:00.000+00:00",
    "authorities": [
        {
            "id": 1,
            "roleCode": "USER",
            "roleDescription": "User role",
            "authority": "USER"
        },
        {
            "id": 2,
            "roleCode": "ADMIN",
            "roleDescription": "Admin role",
            "authority": "ADMIN"
        }
    ]
};

export const prescription = [
    {
        "id": {
            "encounterId": 1,
            "medicineId": 1
        },
        "encounter": {
            "id": 29,
            "date": "2021-11-07",
            "patient": {
                "id": 1009,
                "firstName": "Glenna",
                "lastName": "Reichert",
                "gender": "Female",
                "age": 65,
                "phoneNumber": 9123456788,
                "email": "glenna.reichert@gmail.com",
                "city": "Kolkata",
                "state": "West Bengal",
                "medicalConditions": [
                    {
                        "id": 6,
                        "name": "Tussis (cough)"
                    },
                    {
                        "id": 7,
                        "name": "Pyrexia (fever)"
                    }
                ]
            },
            "user": {
                "id": 1,
                "username": "John",
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "email": "john.cage@gmail.com",
                "enabled": true,
                "role": "Physician",
                "authorities": [
                    {
                        "id": 1,
                        "roleCode": "USER",
                        "roleDescription": "User role",
                        "authority": "USER"
                    },
                    {
                        "id": 2,
                        "roleCode": "ADMIN",
                        "roleDescription": "Admin role",
                        "authority": "ADMIN"
                    }
                ],
                "credentialsNonExpired": true,
                "accountNonExpired": true,
                "accountNonLocked": true
            }
        },
        "medicine": {
            "id": 1,
            "genericName": "Paracetamol",
            "brandName": "Dolo",
            "category": {
                "id": 1,
                "name": "Antipyretics"
            },
            "ingredients": [
                {
                    "id": 1,
                    "name": "Paracetamol"
                }
            ]
        },
        "dosage": "1 time a day"
    }
];

export const allergy = {
    "id": 19,
    "name": "Paracetamol Allergy",
    "reaction": "Tiredness",
    "severity": "MILD",
    "status": "ACTIVE",
    "ingredient": {
        "id": 1,
        "name": "Paracetamol"
    },
    "encounter": {
        "id": 28,
        "date": "2021-11-07",
        "patient": {
            "id": 1009,
            "firstName": "Glenna",
            "lastName": "Reichert",
            "gender": "Female",
            "age": 65,
            "phoneNumber": 9123456788,
            "email": "glenna.reichert@gmail.com",
            "city": "Kolkata",
            "state": "West Bengal",
            "medicalConditions": [
                {
                    "id": 6,
                    "name": "Tussis (cough)"
                },
                {
                    "id": 7,
                    "name": "Pyrexia (fever)"
                }
            ]
        },
        "user": {
            "id": 1,
            "username": "John",
            "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
            "email": "john.cage@gmail.com",
            "enabled": true,
            "role": "Physician",
            "authorities": [
                {
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                    "authority": "USER"
                },
                {
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                    "authority": "ADMIN"
                }
            ],
            "credentialsNonExpired": true,
            "accountNonExpired": true,
            "accountNonLocked": true
        }
    },
    "patient": {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "gender": "Female",
        "age": 65,
        "phoneNumber": 9123456788,
        "email": "glenna.reichert@gmail.com",
        "city": "Kolkata",
        "state": "West Bengal",
        "medicalConditions": [
            {
                "id": 6,
                "name": "Tussis (cough)"
            },
            {
                "id": 7,
                "name": "Pyrexia (fever)"
            }
        ]
    }
};

export const recentPatients = [
    {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-09"
    },
    {
        "id": 1008,
        "firstName": "Nicholas",
        "lastName": "Runolfsdottir",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1021,
        "firstName": "Leroy",
        "lastName": "Blair",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1027,
        "firstName": "Alanna",
        "lastName": "Cantrell",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-07"
    },
    {
        "id": 1016,
        "firstName": "Osbert",
        "lastName": "Valdez",
        "age": 32,
        "gender": "Male",
        "date": "2021-11-02"
    },
    {
        "id": 1024,
        "firstName": "Emmett",
        "lastName": "Marshall",
        "age": 45,
        "gender": "Male",
        "date": "2021-11-02"
    }
];

export const prescriptionData = [
    {
        "brandName": "Tylenol",
        "genericName": "Acetaminophen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Advil",
        "genericName": "Ibuprofen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Dolo",
        "genericName": "Paracetamol",
        "dosage": "1 time a day",
        "date": "2021-10-26"
    }
];

export const medicineList = [
    {
        "id": 1,
        "brandName": "Dolo",
        "cause": "Incompatible with patients' medical conditions"
    },
    {
        "id": 13,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 17,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 22,
        "brandName": "Betadine",
        "cause": ""
    },
    {
        "id": 25,
        "brandName": "Iodex",
        "cause": ""
    },
    {
        "id": 46,
        "brandName": "Desyrel",
        "cause": "Incompatible with patients' medical conditions"
    }
];