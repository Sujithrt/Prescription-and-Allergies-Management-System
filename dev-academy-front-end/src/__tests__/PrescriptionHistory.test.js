import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow } from 'enzyme';
import PrescriptionHistory from '../components/prescriptionHistory/PriscriptionHistory'
import { prescriptionData } from './../testConstants'

describe("PrescriptionHistory Test", () => {
    it("PrescriptionHistory is rendered as expected", () => {
        const prescriptionHistory = shallow(<PrescriptionHistory prescriptionData={prescriptionData} error="" />);
        expect(shallowToJson(prescriptionHistory)).toMatchSnapshot();
    });
    it("Patient has no prescription history", () => {
        const prescriptionHistory = shallow(<PrescriptionHistory prescriptionData={[]} error="Patient does not have any Prescription History"/>);
        expect(shallowToJson(prescriptionHistory)).toMatchSnapshot();
    });
});

describe("PrescriptionHistory Test", () => {
    it("Open PrescriptionHistory Modal Window", () => {
        const prescriptionHistory = shallow(<PrescriptionHistory prescriptionData={prescriptionData} error=""/>);
        prescriptionHistory.find({ 'data-testid': 'viewPrescriptionHistoryButton' }).simulate('click')
    });
    it("Close PrescriptionHistory Modal Window by hide", () => {
        const prescriptionHistory = shallow(<PrescriptionHistory prescriptionData={prescriptionData} error=""/>);
        prescriptionHistory.find({ 'data-testid': 'hideMenu' }).simulate('hide')
    });
    it("Close PrescriptionHistory Modal Window by clicking close button", () => {
        const prescriptionHistory = shallow(<PrescriptionHistory prescriptionData={prescriptionData} error=""/>);
        prescriptionHistory.find({ 'data-testid': 'closeModalButton' }).simulate('click')
    });
});