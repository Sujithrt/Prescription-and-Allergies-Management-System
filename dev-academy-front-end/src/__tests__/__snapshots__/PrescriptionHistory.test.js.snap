// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrescriptionHistory Test Patient has no prescription history 1`] = `
<Fragment>
  <Button
    active={false}
    data-testid="viewPrescriptionHistoryButton"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    variant="primary"
  >
    View Patient Priscription History
  </Button>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    data-testid="hideMenu"
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
    size="lg"
    style={
      Object {
        "zIndex": 2000,
      }
    }
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        Prescription History
      </ModalTitle>
    </ModalHeader>
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "appear": false,
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "render": [Function],
        }
      }
      variant="primary"
    >
      Patient does not have any Prescription History
    </Alert>
    <ModalFooter>
      <Button
        active={false}
        data-testid="closeModalButton"
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
</Fragment>
`;

exports[`PrescriptionHistory Test PrescriptionHistory is rendered as expected 1`] = `
<Fragment>
  <Button
    active={false}
    data-testid="viewPrescriptionHistoryButton"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    variant="primary"
  >
    View Patient Priscription History
  </Button>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    data-testid="hideMenu"
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
    size="lg"
    style={
      Object {
        "zIndex": 2000,
      }
    }
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        Prescription History
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Brand Name
            </th>
            <th>
              Generic Name
            </th>
            <th>
              Dosage
            </th>
            <th>
              Date of Prescription
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="0"
          >
            <td>
              Tylenol
            </td>
            <td>
              Acetaminophen
            </td>
            <td>
              3 times a day
            </td>
            <td>
              2021-10-26
            </td>
          </tr>
          <tr
            key="1"
          >
            <td>
              Advil
            </td>
            <td>
              Ibuprofen
            </td>
            <td>
              3 times a day
            </td>
            <td>
              2021-10-26
            </td>
          </tr>
          <tr
            key="2"
          >
            <td>
              Dolo
            </td>
            <td>
              Paracetamol
            </td>
            <td>
              1 time a day
            </td>
            <td>
              2021-10-26
            </td>
          </tr>
        </tbody>
      </table>
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        data-testid="closeModalButton"
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
</Fragment>
`;
